src/datablock/block.jl

```diff
@@ -155,3 +155,47 @@ function checkblock(
 end
 
 mockblock(block::Keypoints{N}) where N = rand(SVector{N, Float32}, block.sz)
+
+
+# TableRow
+
+"""
+    TableRow{M, N}(catcols, contcols, categorydict) <: Block
+
+`Block` for table rows with M categorical and N continuous columns. `data`
+is valid if it satisfies the `AbstractRow` interface in Tables.jl, values 
+present in indices for categorical and continuous columns are consistent, 
+and `data` is indexable by the elements of `catcols` and `contcols`.
+"""
+struct TableRow{M, N} <: Block
+    catcols
+    contcols
+    categorydict
+end
+
+function TableRow(catcols, contcols, categorydict)
+    TableRow{length(catcols), length(contcols)}(catcols, contcols, categorydict)
+end
+
+function checkblock(block::TableRow, x)
+    (all(col -> x[col] ∈ block.categorydict[col], block.catcols) &&
+    all(col -> x[col] isa Number, block.contcols) &&
+    all(col -> haskey(block.categorydict, col), block.catcols))
```

Comment:
This will error when a column is missing.

Reply:
Should be fixed now.